packages/UI/src/components/Checkbox/Checkbox.tsx

```diff
@@ -5,49 +5,73 @@ import CheckmarkImg from './../../assets/icons/Checkmark';
+      'flex h-6 w-6 items-center justify-center rounded-sm border-2 border-solid border-brand-alt-nebula-500 bg-grayscale-50 focus-within:outline focus-within:outline-2 focus-within:outline-offset-2 focus-within:outline-brand-alt-nebula-500 hover:outline hover:outline-4 hover:outline-offset-0 hover:outline-brand-alt-nebula-100 disabled:border-grayscale-500 disabled:bg-grayscale-200 disabled:hover:outline-none disabled:hover:outline-0 hover:disabled:outline-0 data-checked:bg-brand-alt-nebula-500 data-checked:disabled:bg-grayscale-200',
     indicatorStyle: '',
   },
-  variants: {
-    checked: {
-      true: {
-        rootStyle: 'bg-brand-alt-nebula-500',
-      },
-      false: {
-        indicatorStyle: 'hidden',
-      },
-    },
-  },
+  variants: {},
 });
 
-type OriginalProps = RadixCheckbox.CheckboxProps &
-  VariantProps<typeof checkboxStyles>;
-
-export type CheckboxProps = Omit<OriginalProps, 'checked' | 'disabled'> & {
-  /**
-   * Indicates whether the checkbox is checked
-   */
-  checked?: boolean;
-  /**
-   * Indicates whether the checkbox is disabled
-   */
-  disabled?: boolean;
-};
+export type CheckboxProps = VariantProps<typeof checkboxStyles> &
+  Omit<
+    RadixCheckbox.CheckboxProps,
+    | 'checked'
+    | 'disabled'
+    | 'defaultChecked'
+    | 'onCheckedChange'
+    | 'required'
+    | 'name'
+    | 'value'
+  > & {
+    /**
+     * The controlled checked state of the checkbox. Must be used in conjunction with onCheckedChange.
+     */
+    checked?: RadixCheckbox.CheckboxProps['checked'] & NonNullable<unknown>;
+    /**
+     * When true, prevents the user from interacting with the checkbox.
+     */
+    disabled?: RadixCheckbox.CheckboxProps['disabled'];
+    /**
+     * The checked state of the checkbox when it is initially rendered. Use when you do not need to control its checked state.
+     */
+    defaultChecked?: RadixCheckbox.CheckboxProps['defaultChecked'] &
+      NonNullable<unknown>;
+    /**
+     * Event handler called when the checked state of the checkbox changes.
+     */
+    onCheckedChange?: RadixCheckbox.CheckboxProps['onCheckedChange'];
+    /**
+     * When true, indicates that the user must check the checkbox before the owning form can be submitted.
+     */
+    required?: RadixCheckbox.CheckboxProps['required'];
+    /**
+     * The name of the checkbox. Submitted with its owning form as part of a name/value pair.
+     */
+    name?: RadixCheckbox.CheckboxProps['name'];
+    /**
+     * The value given as data when submitted with a name.
+     */
+    value?: RadixCheckbox.CheckboxProps['value'];
+  };
 
 /**
  * @description Created with the Checkbox component from Radix UI.
  * https://www.radix-ui.com/primitives/docs/components/checkbox
  */
 
-export default function Checkbox({ className, ...props }: CheckboxProps) {
-  const { rootStyle, indicatorStyle } = checkboxStyles({
-    checked: props.checked,
```

Comment:
I removed `checked: props.checked` because we already have `data-checked:` taking its role.
using tailwind variants for this is an over kill as tailwind itself can handle this by targeting the html attributes that are being mutated by radix ui.